.github/workflows/python-tests.yml

```diff
@@ -0,0 +1,29 @@
+name: Python Tests
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.8]
+    steps:
+    - uses: actions/checkout@v4
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v5
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        pip install poetry
+        poetry install
+    - name: Run pytest and generate HTML report
+      run: poetry run pytest --html=report.html --self-contained-html
```

Comment:
You could additionally use [pytest-cov](https://pypi.org/project/pytest-cov/) and then upload the coverage report.

Reply:
yeah, makes sense